irohad/ametsuchi/impl/postgres_wsv_command.cpp

```diff
@@ -106,32 +106,30 @@ namespace iroha {
       const auto perm_str =
           shared_model::interface::GrantablePermissionSet({permission})
               .toBitstring();
-      auto query =
-          (boost::format(
-               "INSERT INTO account_has_grantable_permissions as "
-               "has_perm(permittee_account_id, account_id, permission) VALUES "
-               "(%1%, %2%, %3%) ON CONFLICT (permittee_account_id, account_id) "
-               // SELECT will end up with a error, if the permission exists
-               "DO UPDATE SET permission=(SELECT has_perm.permission | %3% "
-               "WHERE (has_perm.permission & %3%) <> %3%);")
-           % transaction_.quote(permittee_account_id)
-           % transaction_.quote(account_id) % transaction_.quote(perm_str))
+      soci::statement st = sql_.prepare
+          << "INSERT INTO account_has_grantable_permissions as "
+             "has_perm(permittee_account_id, account_id, permission) VALUES "
+             "(:permittee_account_id, :account_id, :perm) ON CONFLICT "
+             "(permittee_account_id, account_id) "
+             // SELECT will end up with a error, if the permission exists
+             "DO UPDATE SET permission=(SELECT has_perm.permission | :perm "
+             "WHERE (has_perm.permission & :perm) <> :perm);";
+      st.exchange(soci::use(permittee_account_id, "permittee_account_id"));
+      st.exchange(soci::use(account_id, "account_id"));
+      st.exchange(soci::use(perm_str, "perm"));
+
+      auto msg =
```

Comment:
Please make that into a lambda. We do not want to instantiate error message in the successful path.